lemon/market_data/instruments.py

```diff
@@ -36,3 +36,36 @@ def get(
             },
         )
         return GetInstrumentsResponse._from_data(resp.json())
+
+    def iter(
+        self,
+        isin: Optional[List[str]] = None,
+        search: Optional[str] = None,
+        type: Optional[List[InstrumentType]] = None,
+        mic: Optional[List[str]] = None,
+        currency: Optional[List[str]] = None,
+        tradable: Optional[bool] = None,
+        sorting: Optional[Sorting] = None,
+        limit: Optional[int] = None,
+        page: Optional[int] = None,
+    ) -> Iterator[Instrument]:
+        while True:
+            resp = self._client.get(
+                "instruments",
+                params={
+                    "isin": isin,
+                    "search": search,
+                    "type": type,
+                    "mic": mic,
+                    "currency": currency,
+                    "tradable": tradable,
+                    "sorting": sorting,
+                    "limit": limit,
+                    "page": page,
+                },
+            )
+            for result in resp.json()["results"]:
+                yield Instrument._from_data(result)
+            if resp.json()["page"] == resp.json()["pages"]:
+                break
+            page = int(resp.json()["page"]) + 1
```

Comment:
We are already outputting the next link in the response, I would directly use it.

Reply:
Then we need to redo the client. Client currently holds the BASE_URL and expects part of the url such as "instruments"

Reply:
Agree with Dennis statement, we don't have to rewrite the client too cause urljoin should take care about proper url concatenation like in picture below

![image](https://user-images.githubusercontent.com/1618754/191211998-19b70159-0c8f-4548-a8b5-a35bee46c774.png)

Reply:
Did not know this! will do it then.
However, to the most important note. This aproach to implement a iter is feasible for us? so i can implement for the other iteratable- endpoints